components/root/root.jsx

```diff
@@ -248,11 +250,16 @@ export default class Root extends React.PureComponent {
             this.onConfigLoaded();
         });
         trackLoadTime();
+
+        window.addEventListener('resize', this.handleWindowResizeEvent);
+        this.props.actions.emitBrowserWindowResized();
```

Comment:
Is this going to cause performance issues? Hitting redux like this for every resize event will cause a lot of stuff to re-render or at least a whole bunch of redux calculations. A resize operation could cause a lot of slowdown.

Reply:
It certainly would, except that the function is throttled on line 290 :)

Reply:
I still kinda like the CSS media query approach because CSS is handled separately from JS by the browser. I understand the concern over having additional markup that is hidden for when in mobile view, but IMO its minimal and given the browser doesn't consider the markup or styling when `display: none;` is used, it's not a huge performance issue. I guess we need to decide what is better for our situation ... again, open to thoughts on this.

Reply:
Also, @harrison and I talked about computed selectors vs. booleans stored in redux and we went with computed selectors for the `isMobileView` etc., but now I'm wondering if it might actually be more performant to store them in redux and only update them when needed instead of computing them based on the dispatched window size updates?

Meaning, if I grab the window size in the action and return a boolean for `isMobileView` based on that size and then get the reducer to decide if it needs to update the store based on whether the value is different, wouldn't that be more performant than updating width/height values and computing the boolean for `isMobileView` each time, thereby forcing component updates more frequently?

Reply:
I agree that this seems like something that should be handled by CSS. Having the browser handle it would remove any performance concerns around redux. Why did we not keep that approach? 

You would have to not even tell redux about the change. If you dispatch an action I believe even it it doesn't effect a state change redux still sends an update to everyone. But maybe in the action you could check the current state and not send the update at all if they are the same?

Reply:
I was thinking about just storing `isMobileView` instead of the window dimensions, but I thought we found some cases which needed the actual dimensions. If those are limited though, we could re-add listeners to those components and have them store the page size in the state. Storing both dimensions and `isMobileView` kind of defeats the purpose of that though.

I don't know if triggering redux updates when resizing is a problem at the moment since resizing should be rare (compared to just using the app), and hopefully the rest of the app is performant enough that that's not a problem (since redux actions are firing all the time). If that is a concern, throttling should hopefully help though.

Reply:
I'm working on a slight alternative along those lines, will push here soon. :)

Reply:
Unrelated to all that (apparently I can't start another thread here), but is calling `emitBrowserWindowResized` on mount to set the initial values in the redux state? I wonder if that delay might cause any issues. Accessing the window in the reducer isn't great, but it would solve any potential delay problems